events/importer/enkora.py

```diff
@@ -1099,35 +1450,86 @@ def _handle_course(self, course: dict) -> dict:
+            },
             "enrolment_start_time": dates["public_reservation_start"],
             "enrolment_end_time": dates["public_reservation_end"],
             "maximum_attendee_capacity": capacity,
             "remaining_attendee_capacity": remaining_capacity,
             "data_source": self.data_source,
             "origin_id": course["reservation_event_group_id"],
             "publisher": self.organization,
-            "location": location_id,
+            "location": location_data,
+            "location_extra_info": location_extra_info,
             "keywords": keywords,
             "in_language": in_language,
+            "offers": offers,
             "images": [],  # must have a list of images, even if there are no images
-            "offers": [],  # must have a list of offers, even if there are no offers
         }
 
-        if location_extra_info:
-            event_data["location_extra_info"] = {"fi": location_extra_info}
+        # Sub-events:
+        # Literally all Enkora courses are multi-event
+        _, sub_events = self.build_sub_events(course, event_data)
+
+        #
+        if self.debug_print_cache_invalidation:
+            print(
+                'curl --header "Content-Type: application/json" https://harrastukset.hel.fi/api/revalidate --data'
+                """ '{{"secret": "", "uri": "/fi/courses/enkora:{}"}}' """.format(
+                    course["reservation_event_group_id"]
+                )
+            )
+
+        return event_data, sub_events
+
+    def build_sub_events(
+        self, course: dict, event_data: dict
+    ) -> tuple[dict, list[dict]]:
+        """
+        Enkora course consists of multiple events
+        :param course: raw Enkora course data from API
+        :param event_data: converted course data suitable for Linked Events storage
+        :return: event_data, altered with sub-events
+        """
 
-        return event_data
+        sub_events = []
+
+        for reservation_event in course["reservation_events"]:
+            # Basis for all sub-events is the super-event
+            sub_event_data = deepcopy(event_data)
+            if "super_event_type" in sub_event_data:
+                del sub_event_data["super_event_type"]
+            sub_event_data["super_event"] = None  # Note: This will be populated on save
+            sub_event_data["start_time"] = timezone.make_aware(
+                reservation_event["time_start"], EnkoraImporter.EEST
+            )
+            sub_event_data["end_time"] = timezone.make_aware(
+                reservation_event["time_end"], EnkoraImporter.EEST
+            )
+            sub_event_data["origin_id"] = "{}_{}".format(
+                course["reservation_event_group_id"],
+                reservation_event["reservation_event_id"],
+            )
+            sub_events.append(sub_event_data)
+
+        event_data["super_event_type"] = Event.SuperEventType.RECURRING
+
+        return event_data, sub_events
```

Comment:
Change: This is a bit confusing. The return value `event_data`, which is a parameter for this method, isn't used anywhere, but it gets mutated. Maybe just don't mutate it here and just return `sub_events`? 🤔

Reply:
It is the responsibility of `build_sub_events()` to make sure super-event and sub-events are both valid and paired properly. Not mutating `event_data` outside this call would break single responsibility -principle.

Additionally: This pattern exists in other importers also.

Keeping this as-is.